### ISSUE TITLE
Library outdated 

### ISSUE THREAD

**Issue by faraz-ahmad-uft** [michaelbromley/ngx-pagination#410]
Has the author stopped supporting this library? Looks like it is not compatible with the latest Angular version.

**Comment by michaelbromley** [comment]
No, it is still maintained - I use it in several projects, some of which are on Angular v17. Which version are you having trouble with?

**Comment by faraz-ahmad-uft** [comment]
Thanks for the response.

I am using Angular v18.

**Comment by michaelbromley** [comment]
ok. please provide all relevant details of any errors you get with v18. I've not tested it yet.

**Comment by godenji** [comment]
Here's one that you'll hit when using standalone components:

> The pipe 'PaginatePipe' appears in 'imports', but is not standalone and cannot be imported directly. It must be imported via an NgModulle

**Comment by michaelbromley** [comment]
@godenji thanks for the info.

So right now you just need to entire `NgxPaginationModule` to your imports array right? I am using this package successfully with standalone components in this way in a v17 app. 

If anyone wants to make the necessary changes to make the lib compatible with standalone, please let me know. Otherwise I'll get around to it when I have time.

**Comment by godenji** [comment]
> So right now you just need to entire NgxPaginationModule to your imports array right? I am using this package successfully with standalone components in this way in a v17 app.

Do you mean adding pagination module  to `importProvidersFrom` of `main.ts`, or adding it to `imports` array of standalone component? If the latter, you cannot import a module within imports section of a standalone component.

Also, on Angular 18 here, so perhaps there are other changes at play; thus the title of this github issue :)

p.s. thank you for creating this useful library.

**Comment by michaelbromley** [comment]
From the [angular.io components docs](https://angular.dev/api/core/Component#imports):

> Standalone components can import other standalone components, directives, and pipes as well as existing NgModules.

Importing the module into your standalone component should work fine. I'm doing it with this lib on a production v17 app.

**Comment by godenji** [comment]
Ah, could be this that I'm hitting:
> This property is only available for standalone components - specifying it for components declared in an NgModule generates a compilation error.

Looks like I need to fully migrate target route and all of its subcomponents to standalone first (I just attempted to convert pagination component in the hierarchy to standalone and import `NgxPaginationModule` in the imports array).

Hopefully it's as straightforward as that...

**Comment by kumarl1** [comment]
@godenji:-  Did you find any alternative solution for this? I am also facing some issue like yours?

**Comment by kumarl1** [comment]
@michaelbromley : Are you planning to migrate it to the latest version(16/17) of the angular?

**Comment by tanver-asthait** [comment]
I have installed in my V18 app importing the NgxPaginationModule in my components import array, and it works fine, a small problem though as the new @for directive doesn't allow me to add paginate pipe alongside it. so I used the old ngFor directive and its working as expected.

**Comment by ali-ic** [comment]
@michaelbromley When you get some time, it would be nice to have the library updated to Angular 18/19. It's currently at Angular 13, so starting to lag behind and feel like it's unsupported since no updates for 2 years. Thanks for all your efforts!

**Comment by michaelbromley** [comment]
If anyone is interested in doing that update, please let me know. That would speed things up.

**Comment by godenji** [comment]
I've updated the project to Angular v19 in this [PR](https://github.com/michaelbromley/ngx-pagination/pull/414).

Standalone version of library components would need to happen in a separate pull request; linked PR above just gets the library itself up-to-date.

**Comment by Harshal-Aubie** [comment]
@godenji 
Can anyone inform us when the support for Angular 19 will roll out?

**Comment by piet23** [comment]
Is there any ETA for the release to support Angular 19 by now, given that the PR is already more than one year old? If not, does anyone know of another library that is more future proof that brings the same functionality? We cannot stick with Angular 18 forever.
